src/WebServer.Logging/ILogger.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+
+namespace Restup.WebServer.Logging
+{
+    public interface ILogger
```

Comment:
```cs
void Info(Func<string> messageGenerator) 
{
  if(!IsInfoEnabled())
    return;

  Info(messageGenerator());
}
```

This will prevent performance hits like this:
```cs
_logger.Info($"Statistics, average response time: {CalculateAvgResponse()}");
```
Even though it might never be actually logged (because log level is set to error), that ```CalculateAvgResponse``` is always executed. You'll solve that with a ```Func<string>``` as it's only executed when it's necessary.